core/src/main/java/com/orientechnologies/orient/core/metadata/security/OPropertyAccess.java

```diff
@@ -0,0 +1,23 @@
+package com.orientechnologies.orient.core.metadata.security;
+
+import com.orientechnologies.orient.core.record.impl.ODocument;
+
+import java.util.Set;
+
+public class OPropertyAccess {
+
+  private Set<String> filtered;
+
+  public OPropertyAccess(ODocument document, OSecurityInternal security) {
+    filtered = security.getFilteredProperties(document);
+  }
+
+  public OPropertyAccess(Set<String> filtered) {
+    this.filtered = filtered;
+  }
+
+  public boolean isReadable(ODocument document, String property) {
```

Comment:
IMHO this is extremely confusing. The relationship between the objects is implicit and it makes it terribly error prone (eg. use the wrong OPropertyAccess for an ODocument).
I'd prefer to have something linke

```
doc.getPropertyAccess().isReadable("property")
```

And to avoid the constructor that uses the security (ie. the security should build the OPropertyAccess)

Reply:
So you mean that we should have na OPropertyAccess that works for all the documents ?

Reply:
No, I mean the opposite.
If you want to have a pre-populated bean that contains access rules, it should be stored in the document, and not built on-the-fly: 

```
doc.getPropertyAccess().isReadable("property")

instead of 

x = new OPropertyAccess(doc, security);
x.isReadable(doc2, "property"); //see doc vs doc2

```
If you want to create it every time you use it, then you should ask for it to the security:

```
security.getPropertyAccess(doc).isReadable("property")
```

Reply:
oh yep, i definitely think we can remove the parameter from `isReadable` we can work around in case we need the doc in isReadable to keep inside the one you have in the constructor